tests/e2e/creation_test.go

```diff
@@ -351,3 +358,112 @@ func (tc *DSCTestCtx) UpdateRegistriesNamespace(targetNamespace, expectedValue s
 		WithCustomErrorMsg("Failed to update RegistriesNamespace to %s, expected %s", targetNamespace, expectedValue),
 	)
 }
+
+// ValidateComponentsDeploymentFailure simulates component deployment failure using restrictive resource quota.
+func (tc *DSCTestCtx) ValidateComponentsDeploymentFailure(t *testing.T) {
+	t.Helper()
+
+	var allComponents = []string{
+		"dashboard", "workbenches", "modelregistry", "ray", "kueue",
+		"codeflare", "trainingoperator", "datasciencepipelines",
+		"kserve", "modelmeshserving", "trustyai", "feastoperator", "llamastackoperator",
+	}
+
+	originalDSC := tc.FetchDataScienceCluster()
+	restrictiveQuota := createRestrictiveQuotaForOperator(tc.AppsNamespace)
+
+	t.Log("Creating restrictive resource quota in operator namespace")
+	tc.EnsureResourceCreatedOrPatched(WithObjectToCreate(restrictiveQuota))
+
+	t.Log("Enabling all components in DataScienceCluster")
+	tc.EnsureResourceCreatedOrPatched(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(updateAllComponentsTransform(allComponents, operatorv1.Managed)),
+	)
+
+	t.Log("Verifying DSC reports all failed components due to restrictive quota")
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithCondition(jqAllComponentsFailureMatcher(allComponents)),
+		WithCustomErrorMsg("DSC should report all components failed due to insufficient quota"),
+		WithEventuallyTimeout(tc.TestTimeouts.mediumEventuallyTimeout),
+	)
+
+	t.Log("Cleaning up restrictive quota")
+	tc.DeleteResource(WithObjectToCreate(restrictiveQuota))
+
+	t.Log("Restoring original DSC state")
+	tc.restoreDSC(t, originalDSC)
+
+	t.Log("Verifying DSC reports components ready after restoring")
+	tc.verifyComponentsReady(t)
+
+	t.Log("Component deployment failure validation test completed successfully")
+}
+
+// Helper function to enable/disable all components.
+func updateAllComponentsTransform(components []string, state operatorv1.ManagementState) testf.TransformFn {
+	transformParts := make([]string, len(components))
+	for i, component := range components {
+		transformParts[i] = fmt.Sprintf(`.spec.components.%s.managementState = "%s"`, component, state)
+	}
+
+	return testf.Transform(strings.Join(transformParts, " | "))
+}
+
+func jqAllComponentsFailureMatcher(components []string) *jq.Matcher {
+	tests := make([]string, 0, len(components))
+	for _, c := range components {
+		tests = append(tests, fmt.Sprintf(`test("%s"; "i")`, c))
+	}
+	return jq.Match(fmt.Sprintf(`
+		.status.conditions[]
+		| select(.type == "ComponentsReady" and .status == "False")
+		| .message
+		| (%s)
+	`, strings.Join(tests, " and ")))
```

Comment:
Are we sure that all components an are failing because of the corresponding deployment not ready?

Reply:
jqAllComponentsFailureMatcher checks all the component names in the error message. do i need to check the deployments also?

Reply:
Yes, we need to make sure each component fails due to its deployment. If it fails for other reasons, the test isn’t valid.

Reply:
Added an event message to confirm deployment failure.